### PR TITLE
Add: Exception option for `no-extend-native` and `no-native-reassign` (f...

### PR DESCRIPTION
**Pull request by gyandeeps**
...ixes #2355)

### CODE REVIEW

File: docs/rules/no-extend-native.md [hunk 1]
```diff
@@ -36,6 +36,13 @@ It *does not* check for any of the following less obvious approaches:
 * `with(Array) { prototype.thing = 'thing'; };`
 * `window.Function.prototype.bind = 'tight';`
 
+## Options
+
+### exceptions
+
+Array of native object names that are permitted to be extended.
```
Comment (nzakas) [inline docs/rules/no-extend-native.md]: Can you show an example here? Same with the other rule.

File: lib/rules/no-extend-native.js [hunk 2]
```diff
@@ -21,6 +21,13 @@ var BUILTINS = [
 
 module.exports = function(context) {
 
+    var config = context.options[0] || {};
+    var exceptions = config.exceptions || [];
+
+    var modifiedBuiltins = BUILTINS.filter(function(builtIn) {
```
Comment (nzakas) [inline lib/rules/no-extend-native.js]: You can optimize a bit by not doing this if `exceptions.length === 0`

File: lib/rules/no-native-reassign.js [hunk 3]
```diff
@@ -11,24 +11,29 @@
 
 module.exports = function(context) {
 
-    var nativeObjects = ["Array", "Boolean", "Date", "decodeURI",
+    var NATIVEOBJECTS = ["Array", "Boolean", "Date", "decodeURI",
```
Comment (nzakas) [inline lib/rules/no-native-reassign.js]: NATIVE_OBJECTS

File: lib/rules/no-extend-native.js [hunk 4]
```diff
@@ -21,6 +21,13 @@ var BUILTINS = [
...
+        return exceptions.indexOf(builtIn) === -1 ? true : false;
```
Comment (nzakas) [inline lib/rules/no-extend-native.js]: You don't need `? true : false ` on this line. The expression already evaluate to true or false.

**Comment by gyandeeps** [comment]
@nzakas All the req

**Comment by nzakas** [comment]
Lgtm

**Pull request merged**